backend/src/services/trading_service.py

```
@@ -216,16 +205,30 @@ async def execute_sell_order(self, token_address: str, amount_tokens: float, sli
+        """
+        Polls for transaction confirmation.
+        """
+        start_time = datetime.now().timestamp()
+        while (datetime.now().timestamp() - start_time) < timeout:
+            try:
+                response = await self.solana_client.get_signature_statuses([signature])
+                if response.value and response.value[0]:
+                    status = response.value[0]
+                    if status.confirmations is not None or status.err is None:
+                        return True
+            except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)